pkg/sql/distsql_physical_planner_test.go

```diff
@@ -664,88 +675,166 @@ func TestDistSQLUnavailableHosts(t *testing.T) {
 			t.Error(err)
 		}
 	})
-}
 
-func TestDistSQLDrainingHosts(t *testing.T) {
-	defer leaktest.AfterTest(t)()
-	defer log.Scope(t).Close(t)
+	drainServer := func(t *testing.T, server serverutils.ApplicationLayerInterface, idx int) {
+		c := server.GetAdminClient(t)
+		req := &serverpb.DrainRequest{DoDrain: true}
+		stream, err := c.Drain(ctx, req)
+		require.NoError(t, err)
+		_, err = stream.Recv()
+		require.NoError(t, err)
+		unexpected, err := stream.Recv()
+		if err != io.EOF {
+			if unexpected != nil {
+				t.Fatalf("unexpected additional response: %v // %v", unexpected, err)
+			}
+			if err == nil {
+				err = errors.New("unexpected response")
+			}
+			t.Fatal(err)
+		}
+	}
 
-	const numNodes = 2
-	tc := serverutils.StartCluster(
-		t,
-		numNodes,
-		base.TestClusterArgs{
-			ReplicationMode: base.ReplicationManual,
-			ServerArgs:      base.TestServerArgs{Knobs: base.TestingKnobs{DistSQL: &execinfra.TestingKnobs{DrainFast: true}}, UseDatabase: "test"},
-		},
-	)
-	ctx := context.Background()
-	defer tc.Stopper().Stop(ctx)
+	t.Run("draining-nodes-in-single-tenant-mode", func(t *testing.T) {
+		skip.UnderDuress(t, "takes 20s")
```

Comment:
I think you can move this skip to the overall test rather that putting it in each subtest.